packages/cli/src/server/output.rs

```diff
@@ -1,157 +1,518 @@
+        }
+
+        Ok(())
+    }
+
+    pub fn push_log(&mut self, platform: Platform, message: BuildMessage) {
+        let snapped = self.is_snapped(platform);
+
+        self.build_logs
+            .get_mut(&platform)
+            .unwrap()
+            .messages
+            .push(message);
+
+        let log = self.build_logs.get(&platform).unwrap();
+        if snapped {
+            self.scroll = log.messages.len().saturating_sub(self.term_height as usize) as u16;
+        }
+    }
+
+    fn is_snapped(&self, platform: Platform) -> bool {
+        let Some(log) = self.build_logs.get(&platform) else {
+            return false;
+        };
+
+        let prev_scrol = log.messages.len().saturating_sub(self.term_height as usize) as u16;
+        prev_scrol == self.scroll
     }
 
-    let mut profile = if config.release { "Release" } else { "Debug" }.to_string();
-    if config.custom_profile.is_some() {
-        profile = config.custom_profile.as_ref().unwrap().to_string();
+    pub fn new_build_logs(&mut self, platform: Platform, update: UpdateBuildProgress) {
+        let snapped = self.is_snapped(platform);
+
+        self.build_logs.entry(platform).or_default().update(update);
+
+        let log = self.build_logs.get(&platform).unwrap();
+        if snapped {
+            self.scroll = log.messages.len().saturating_sub(self.term_height as usize) as u16;
+        }
     }
-    let hot_reload = if config.hot_reload { "RSX" } else { "Normal" };
-    let crate_root = crate_root().unwrap();
-    let custom_html_file = if crate_root.join("index.html").is_file() {
-        "Custom [index.html]"
-    } else {
-        "None"
-    };
-    let url_rewrite = if config.dioxus_config.web.watcher.index_on_404 {
-        "True"
-    } else {
-        "False"
-    };
-
-    let proxies = &config.dioxus_config.web.proxy;
-
-    if options.changed.is_empty() {
-        println!(
-            "{} @ v{} [{}]",
-            "Dioxus".bold().green(),
-            clap::crate_version!(),
-            chrono::Local::now().format("%H:%M:%S").to_string().dimmed()
-        );
-    } else {
-        println!(
-            "Project Reloaded: {}\n",
-            format!(
-                "Changed {} files. [{}]",
-                options.changed.len(),
-                chrono::Local::now().format("%H:%M:%S").to_string().dimmed()
-            )
-            .purple()
-            .bold()
-        );
+
+    pub fn new_ready_app(&mut self, build_engine: &mut Builder, results: Vec<BuildResult>) {
+        for result in results {
+            let (stdout, stderr) = build_engine
+                .children
+                .iter_mut()
+                .find_map(|(platform, child)| {
+                    if platform == &result.platform {
+                        Some((child.stdout.take().unwrap(), child.stderr.take().unwrap()))
+                    } else {
+                        None
+                    }
+                })
+                .unwrap();
```

Comment:
When starting up with `--platform web` this line panicks.
thread 'main' panicked at packages\cli\src\server\output.rs:276:18:
called `Option::unwrap()` on a `None` value

Worked around it with this.
```rust
match build_engine
    .children
    .iter_mut()
    .find_map(|(platform, child)| {
        if platform == &result.platform {
            Some((child.stdout.take(), child.stderr.take()))
        } else {
            None
        }
    }) {
    Some((Some(stdout), Some(stderr))) => {
        let platform = result.platform.clone();

        let app = RunningApp {
            result,
            stdout: BufReader::new(stdout).lines(),
            stderr: BufReader::new(stderr).lines(),
            stdout_line: String::new(),
            stderr_line: String::new(),
        };

        self.running_apps.insert(platform, app);
    },
    _ => {
        // Handle the None case or continue the loop
        continue;
    }
}
```

Reply:
Good catch! Stdout is not guaranteed for web builds - we want to actually pipe the output to the terminal. I've updated the code to do that.